From 9c329421f99f38f1e4dfbb01bc802514d92296e1 Mon Sep 17 00:00:00 2001
From: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Date: Fri, 9 Mar 2018 17:58:24 +0700
Subject: [PATCH 707/909] arm64: dts: r8a7795: Change IPMMU-MM and IPMMU-caches
 order in DT

commit edc0cc60fc555c7c4d8a5498515e0e28e757517d from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

In current implementation, IPMMU-caches will defer probe for waiting
IPMMU-MM by returning -EPROBE_DEFER. However, it leads to other driver,
e.g. DMAC, which rely on IPMMU-caches to setup its configuration,
defer probe as well.

As a workaround for this issue, reorder IPMMU-MM and IPMMU-caches in DT
in order to make IPMMU-MM probe first since current driver only parses DT
info when setting up IPMMU domain.

Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7795.dtsi | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795.dtsi b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
index 24efa1f59ee2..80ef8f80be65 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
@@ -1461,6 +1461,16 @@
 			       <&ipmmu_ds1 30>, <&ipmmu_ds1 31>;
 		};
 
+		ipmmu_mm: mmu@e67b0000 {
+			compatible = "renesas,ipmmu-r8a7795";
+			reg = <0 0xe67b0000 0 0x1000>;
+			interrupts = <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
 		ipmmu_ds0: mmu@e6740000 {
 			compatible = "renesas,ipmmu-r8a7795";
 			reg = <0 0xe6740000 0 0x1000>;
@@ -1497,16 +1507,6 @@
 			status = "disabled";
 		};
 
-		ipmmu_mm: mmu@e67b0000 {
-			compatible = "renesas,ipmmu-r8a7795";
-			reg = <0 0xe67b0000 0 0x1000>;
-			interrupts = <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
-			#iommu-cells = <1>;
-			status = "disabled";
-		};
-
 		ipmmu_mp0: mmu@ec670000 {
 			compatible = "renesas,ipmmu-r8a7795";
 			reg = <0 0xec670000 0 0x1000>;
-- 
2.17.1

